test/common/grafana_routes.go

```diff
@@ -0,0 +1,110 @@
+	httpClient, err := buildHTTPClientFromContext(ctx)
+	if err != nil {
+		t.Fatal("failed to create test http client", err)
+	}
+	//reconcile idp setup
+	if err := createTestingIDP(context.TODO(), ctx.Client, httpClient, ctx.SelfSignedCerts); err != nil {
+		t.Fatal("failed to reconcile testing idp", err)
+	}
+	grafanaRootHostname, err := getGrafanaRoute(ctx.Client)
+	if err != nil {
+		t.Fatal("failed to get grafana route", err)
+	}
+	//perform a request that we expect to be forbidden initially
+	forbiddenResp, err := httpClient.Get(grafanaRootHostname)
+	if err != nil {
+		t.Fatal("failed to perform expected forbidden request", err)
+	}
+	if forbiddenResp.StatusCode != http.StatusForbidden {
+		t.Fatalf("unexpected status code on forbidden request, got=%+v", forbiddenResp)
+	}
+	//retrieve an openshift oauth proxy cookie
+	grafanaOauthHostname := fmt.Sprintf("%s/oauth/start", grafanaRootHostname)
+	if err := resources.DoAuthOpenshiftUser(grafanaOauthHostname, grafanaCredsUsername, grafanaCredsPassword, httpClient); err != nil {
+		t.Fatal("failed to login through openshift oauth proxy", err)
+	}
+	successResp, err := httpClient.Get(grafanaRootHostname)
+	if err != nil {
+		t.Fatal("failed to perform test request to grafana", err)
+	}
+	defer successResp.Body.Close()
+	if successResp.StatusCode != http.StatusOK {
+		t.Fatalf("unexpected status code on success request, got=%+v", successResp)
+	}
+}
+
+func TestGrafanaExternalRouteDashboardExist(t *testing.T, ctx *TestingContext) {
+	httpClient, err := buildHTTPClientFromContext(ctx)
+	if err != nil {
+		t.Fatal("failed to create test http client", err)
+	}
+	//reconcile idp setup
+	if err := createTestingIDP(context.TODO(), ctx.Client, httpClient, ctx.SelfSignedCerts); err != nil {
+		t.Fatal("failed to reconcile testing idp", err)
+	}
+	grafanaRootHostname, err := getGrafanaRoute(ctx.Client)
+	if err != nil {
+		t.Fatal("failed to get grafana route", err)
+	}
+	//retrieve an openshift oauth proxy cookie
+	grafanaOauthHostname := fmt.Sprintf("%s/oauth/start", grafanaRootHostname)
+	if err = resources.DoAuthOpenshiftUser(grafanaOauthHostname, grafanaCredsUsername, grafanaCredsPassword, httpClient); err != nil {
+		t.Fatal("failed to login through openshift oauth proxy", err)
+	}
+	//get dashboards for grafana from the external route
+	grafanaDashboardsUrl := fmt.Sprintf("%s/api/search", grafanaRootHostname)
+	dashboardResp, err := httpClient.Get(grafanaDashboardsUrl)
+	if err != nil {
+		t.Fatal("failed to perform test request to grafana", err)
+	}
+	defer dashboardResp.Body.Close()
+	if dashboardResp.StatusCode != http.StatusOK {
+		t.Fatalf("unexpected status code on success request, got=%+v", dashboardResp)
+	}
+	var dashboards []interface{}
+	if err := json.NewDecoder(dashboardResp.Body).Decode(&dashboards); err != nil {
+		t.Fatal("failed to decode grafana dashboards response", err)
+	}
+	if len(dashboards) == 0 {
+		t.Fatal("no grafana dashboards returned from grafana api")
```

Comment:
There is an existing test for checking the actual dashboards. This is effectively just testing that the response back from the exposed Grafana route is valid.